test/benchmarks/material/table/basic-table.ts

```diff
@@ -0,0 +1,37 @@
+import {Component, Input} from '@angular/core';
+
+/**
+ * Just a generic object whose keys are strings and whose values can be anything.
+ *
+ * This is used by BasicTable whose rows are objects whose keys correspond to the cols provided and
+ * whose values can be anything.
+ */
+export interface GenericObject {
+  [key: string]: any;
+}
+
+/**
+ * @title A basic table.
+ * @input rows - Objects whose properties are the cols provided.
+ * @input cols - An array of strings.
+ */
+@Component({
+  selector: 'basic-table',
+  template: `
+  <table mat-table [dataSource]="rows" class="mat-elevation-z8">
+
+    <ng-container *ngFor="let col of cols" matColumnDef="{{ col }}">
```

Comment:
I think we should use static columns instead of dnamically creating them with `ngFor`